internal/exec/stages/disks/luks.go

```diff
@@ -224,3 +302,44 @@ func (s *stage) createLuks(config types.Config) error {
 
 	return nil
 }
+
+func (s *stage) checkLuksDeviceExists(luks types.Luks, keyFilePath string) (bool, error) {
+	devAlias := execUtil.DeviceAlias(*luks.Device)
+	if _, err := s.Logger.LogCmd(
+		exec.Command(distro.CryptsetupCmd(), "isLuks", "--type", "luks2", devAlias),
+		"checking if %v is a luks device", *luks.Device,
+	); err != nil {
+		// isLuks returns exit status 1 if the device is not LUKS
+		return false, nil
+	}
+
+	if luks.Label != nil {
+		fsInfo, err := execUtil.GetFilesystemInfo(devAlias, true)
+		if err != nil {
+			return false, fmt.Errorf("retrieving filesystem info: %v", err)
+		}
+		if fsInfo.Label != *luks.Label {
+			return false, nil
+		}
+	}
+
+	if luks.UUID != nil {
+		uuid, err := exec.Command(distro.CryptsetupCmd(), "luksUUID", devAlias).CombinedOutput()
+		if err != nil {
+			return false, err
+		}
+		if strings.TrimSpace(string(uuid)) != *luks.UUID {
+			return false, nil
+		}
+	}
+
+	// open the device to make sure the keyfile is valid
+	if _, err := s.Logger.LogCmd(
+		exec.Command(distro.CryptsetupCmd(), "luksOpen", devAlias, luks.Name, "--key-file", keyFilePath),
+		"opening luks device %v", luks.Name,
+	); err != nil {
+		return false, nil
+	}
+	s.Logger.Debug("device %v already exists, reusing", luks.Name)
```

Comment:
Should remove this. In the case `wipeVolume` is `true`, it's not true and in the case it's `false` case, we already print something similar in the caller.

Reply:
Ah yeah, good point.

Reply:
Updated, after looking at it again I realized that there was a bug (if `wipeVolume` was `true` on a device that could be re-used it would fail on the `wipefs` because the device was opened), so I went moved the check for if the device exists & the FilesystemInfo fetch inside of the `WipeVolume` check.